airflow/jobs/local_task_job.py

```diff
@@ -162,11 +162,7 @@ def heartbeat_callback(self, session=None):
                 "Terminating instance.",
                 ti.state
             )
-            if ti.state == State.FAILED and ti.task.on_failure_callback:
-                context = ti.get_template_context()
-                ti.task.on_failure_callback(context)
-            if ti.state == State.SUCCESS and ti.task.on_success_callback:
-                context = ti.get_template_context()
-                ti.task.on_success_callback(context)
```

Comment:
In case of success, do I correctly understand that now `task.on_success_callback` will be executed only in `_run_raw_task` via `_run_success_callback`?

Reply:
that's correct, callbacks should only be executed within _run_raw_task to avoid race condiitions.